### ISSUE TITLE
ImListTrans.concat()

### ISSUE THREAD

**Issue by pniederw** [GlenKPeterson/UncleJim#12]
I think `ImListTrans` should override `concat`, refining the return type to `ImListTrans`.

PS: Thanks for transient collections. Big improvement for my use case.

**Comment by GlenKPeterson** [comment]
Fixed in 2.0.7.  Also union() for sets had the same issue which is taken care of.  You are probably the only person to use last night's build, so I'm going to correct the other places where I wrote this and just tell you that I renamed some of the stuff I just made public:

* ImUnsortMap is now ImUnsortedMap
* ImUnsortSet is now ImUnsortedSet
* ImListTrans is now MutableList
* ImUnsortMapTrans is now MutableUnsortedMap
* ImUnsortSetTrans is now MutableUnsortedSet
* The `persistent()` method on all of those interfaces is now `immutable()`
* The `asTransient()` method on all of those interfaces is now `mutable()`

Justification (such as it is) for these changes is in the latest CHANGE_LOG.  Thanks for living on the bleeding edge.  Sorry about this.  At some point, I'll have to be more conservative about releases and changes.

**Comment by GlenKPeterson** [comment]
Peter,

I can't tell you how nice it is to make a release and have it tested (or at least tried out) the next day.  Your bug reports and comments have been really helpful.  I wanted you to know that I appreciate your work!  Thank you.

Sincerely,
Glen

**Comment by pniederw** [comment]
Glen, it's the least I can do. Thanks for this great project.

I'm running into a problem with 2.0.7:

```
java.lang.ClassCastException: org.organicdesign.fp.collections.PersistentVector cannot be cast to org.organicdesign.fp.collections.MutableList
	at org.organicdesign.fp.collections.MutableList.concat(MutableList.java:17)
```

It seems that `MutableList.concat` now calls `ImList.concat`, which returns an immutable list. Therefore, casting to `MutableList` in `MutableList.concat` fails.

**Comment by GlenKPeterson** [comment]
Sorry about that.  Fix is in 2.0.8, available now.  I made sure to write a test for this, prove that the test showed the error, then prove that it's fixed.

**Comment by pniederw** [comment]
2.0.8 works fine. Thanks.
